packages/patternfly-4/react-core/src/components/Select/SelectMenu.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Select/select';
+import { default as formStyles } from '@patternfly/react-styles/css/components/Form/form';
+import { css } from '@patternfly/react-styles';
+import { SelectConsumer, SelectVariant } from './selectConstants';
+
+// tslint:disable-next-line
+const FocusTrap: any = require('focus-trap-react');
+
+export interface SelectMenuProps {
+  /** Content rendered inside the SelectMenu */
+  children: React.ReactElement[];
+  /** Additional classes added to the SelectMenu control */
+  className?: string;
+  /** Flag indicating the Select is expanded */
+  isExpanded: boolean;
+  /** Flag indicating the Select options are grouped */
+  isGrouped?: boolean;
+  /** Currently selected option (for single, typeahead variants) */
+  selected?: string | string[];
+  /** Currently checked options (for checkbox variant) */
+  checked?: string[];
+  /** Internal flag for specifiying how the menu was opened */
+  openedOnEnter?: boolean;
+  /** Internal callback for ref tracking */
+  sendRef?: (ref: React.ReactNode, index: number) => void;
+  /** Internal callback for keyboard navigation */
+  keyHandler?: (index: number, position: string) => void;
+}
+
+export class SelectMenu extends React.Component<SelectMenuProps> {
```

Comment:
I think this can also be a React.FunctionComponent since there is no state.

Reply:
and there are no lifecycle methods here, these methods can be defined in the body of the functionComponent